### PR TITLE
refactor(ivy): migrate component ref spec from render

### PR DESCRIPTION
**Pull request by cexbrayat**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?

Migrates `component_ref_spec.ts` from `render3` to `acceptance`.
Some tests are focused on Renderer3 and have been marked as `onlyInIvy`.
Also note that the `Injector.create` calls are now using the non-deprecated version.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/test/acceptance/component_ref_spec.ts [hunk 1]
```diff
@@ -6,108 +6,112 @@
...
+     * to be used with `ComponentFactoryResolver`
+     * @param componentType The component to add to the TestingModule
+     */
+    function setupModuleForViewEngineTest<T>(componentType: Type<T>) {
+      @NgModule({entryComponents: [componentType], declarations: [componentType]})
+      class TestModule {
+      }
+
+      TestBed.configureTestingModule({imports: [TestModule]});
+    }
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/component_ref_spec.ts]: Personally I would keep it without a flag and setup the same testing module for VE and ivy. I'm pretty sure that we will still need some changes to switch to the "don't need entry components" mode (documentation, as the very minimum) so for now I would just try to be on par with VE as much as possible. As soon as ivy is rolled out to the public we can take advantage of its super-powers and start simplifying things.

tl;dr; for now I would try to keep things consistent with VE as much as possible.

File: packages/core/test/acceptance/component_ref_spec.ts [hunk 2]
```diff
@@ -6,108 +6,112 @@
...
 
+      if (!ivyEnabled) {
+        setupModuleForViewEngineTest(TestComponent);
+      }
+
+      const cfr = TestBed.inject(ComponentFactoryResolver);
       const cf = cfr.resolveComponentFactory(TestComponent);
 
-      expect(cf.selector).toBe('test');
+      expect(cf.selector).toBe('test[foo],bar');
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/component_ref_spec.ts]: Hmmm, this is wired, I would expect VE and ivy to behave the same here. I don't remember why we've got `cf.selector` but probably used to create a host element of the dynamically created component. Could you verify? 

Looking at the public API (https://angular.io/api/core/ComponentFactory#selector) it should be a full selector but again, I guess it is only used for the host element creation. Again, could you verify?

If my guess is correct I _think_ that we should either:
* align with VE (if possible);
* turn this tests that verifies a tag name of the dynamically created component (so have it a more integration test) and mark this thing as a braking change (for me this is more like an implementation detail and factories should go away as soon as VE is removed).

File: packages/core/test/acceptance/component_ref_spec.ts [hunk 3]
```diff
@@ -6,108 +6,112 @@
...
       expect(cf.outputs).toEqual([
         {propName: 'out1', templateName: 'out1'},
-        {propName: 'out2', templateName: 'output-attr-2'},
+        {propName: 'out2', templateName: 'output-attr-2'}
       ]);
     });
   });
 
-  describe('create()', () => {
+  onlyInIvy('renderer3 is available only in Ivy').describe('create()', () => {
```
[the remark this answers is not in the document]
Comment (pkozlowski-opensource) [reply packages/core/test/acceptance/component_ref_spec.ts]: So we are still debating the exact shape / interface of `Renderer3` and for sure this is the ivy-only concept. My suggestion for now would be to keep this test as hand-written and don't migrate it to acceptance. We still need to figure out long-term `Renderer` interface story (probably will happen soon, as part of the size work)
